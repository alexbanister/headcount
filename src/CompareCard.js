import React from 'react';
import { VictoryChart, VictoryLine, VictoryAxis } from 'victory';
import PropTypes from 'prop-types';

const CompareCard = ( {compareInfo} ) => {
  return (
    <div className="card">
      <h2 className="district-name">{compareInfo.firstDistrict.title}</h2>
      <div className="card-data">{compareInfo.firstDistrict.average}</div>
      <h1>{compareInfo.difference.compared}</h1>
      <VictoryChart>
        <VictoryAxis
          dependentAxis
          tickFormat={(xValue) => (`${xValue * 100}%`)}
        />
        <VictoryLine
          interpolation='monotoneX'
          u={Object.values(compareInfo.firstDistrict.districtData.districtData)}
          style={{
            data: {stroke: "#EE7B2A" }
          }}
        />
        <VictoryLine
          interpolation='monotoneX'
          data={
            Object.values(compareInfo.secondDistrict.districtData.districtData)}
          style={{
            data: {stroke: "#349BF5" }
          }}
        />
      </VictoryChart>
      <div className="card-data">{compareInfo.secondDistrict.average}</div>
      <h2 className="district-name">{compareInfo.secondDistrict.title}</h2>
    </div>
  );
};

CompareCard.propTypes = {
  compareInfo: PropTypes.objectOf(PropTypes.object)
};

export default CompareCard;
