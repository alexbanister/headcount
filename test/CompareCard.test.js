import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CompareCard from '../src/CompareCard';

describe('CompareCard', () => {

  it('should create an instance of a card', () => {
    const dummyCompare = {
      firstDistrict: {
        title: 'Colorado',
        average: .8
      },
      secondDistrict: {
        title: 'Turing',
        average: .9
      },
      difference: {
        compare: .85
      }
    };

    const wrapper = shallow(<CompareCard compareInfo={dummyCompare}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
