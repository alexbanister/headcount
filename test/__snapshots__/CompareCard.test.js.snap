exports[`CompareCard should create an instance of a card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    compareInfo={
        Object {
            "difference": Object {
              "compare": 0.85,
            },
            "firstDistrict": Object {
              "average": 0.8,
              "title": "Colorado",
            },
            "secondDistrict": Object {
              "average": 0.9,
              "title": "Turing",
            },
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="district-name">
          Colorado
</h2>,
        <div
          className="card-data">
          0.8
</div>,
        <h1 />,
        <div
          className="card-data">
          0.9
</div>,
        <h2
          className="district-name">
          Turing
</h2>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Colorado",
          "className": "district-name",
        },
        "ref": null,
        "rendered": "Colorado",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": 0.8,
          "className": "card-data",
        },
        "ref": null,
        "rendered": 0.8,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": 0.9,
          "className": "card-data",
        },
        "ref": null,
        "rendered": 0.9,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Turing",
          "className": "district-name",
        },
        "ref": null,
        "rendered": "Turing",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="district-name">
            Colorado
</h2>,
          <div
            className="card-data">
            0.8
</div>,
          <h1 />,
          <div
            className="card-data">
            0.9
</div>,
          <h2
            className="district-name">
            Turing
</h2>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Colorado",
            "className": "district-name",
          },
          "ref": null,
          "rendered": "Colorado",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": 0.8,
            "className": "card-data",
          },
          "ref": null,
          "rendered": 0.8,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": 0.9,
            "className": "card-data",
          },
          "ref": null,
          "rendered": 0.9,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
            "className": "district-name",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
